Jenkinsfile

```diff
@@ -1,11 +1,46 @@
+properties([
+    buildDiscarder(logRotator(numToKeepStr: '10')),
+    parameters([
+        booleanParam(defaultValue: false,
+            description: 'Override distfeeds.conf links',
+            name: 'OVERRIDE_PACKAGES'),
+        stringParam(defaultValue: '',
+            description: 'Works only with OVERRIDE_PACKAGES set to true.\
+                          Path to a directory containing the packages folder,\
+                          if null defaults to ${JOB_PATH}/artifact/bin/pistachio',
+            name: "PACKAGES_PATH"),
+        stringParam(defaultValue: '',
+            description: 'Path to the job that needs to be tested, if null downloads newest stable release',
+            name: "JOB_PATH"),
+    ])
+])
+
 node('boardfarm') {
     stage("Configure") {
         deleteDir()
 
+        def image_path = "http://downloads.creatordev.io/openwrt/ci40-v1.1.0/pistachio/marduk"
+        def image_name = "openwrt-v1.1.0-pistachio-marduk-squashfs-factory.ubi"
+
+        if(params.JOB_PATH.trim()) {
+            /* Since it's impossible to use in-built Groovy functions due to sandboxing this little script
+               uses Jenkins XML api to get a list of artifacts and greps between the XML tags for the name of
+               ubi file. */
+            image_name = sh (
+                script: "curl -v --silent \
+                    '${params.JOB_PATH}/api/xml' \
+                    2>/dev/null | grep -o -E 'openwrt[^>]+.ubi' | head -n 1",
+                returnStdout: true).trim()
+            image_path = params.JOB_PATH + "/artifact/bin/pistachio"
+        }
+
+        def dl_path = "${image_path}/${image_name}"
+        sh "wget '${dl_path}' -O ${env.WEBSERVER_PATH}/openwrt.ubi"
```

Comment:
I'd consider that a different ticket altogether, this one mainly focuses on changing existing functionality to allow opkg tests, I don't want to bloat it with adding extra functionality. We don't have sysupgrade tests and if we want to fully accommodate that it needs quite a bit of work.